sql-plugin/src/main/scala/com/nvidia/spark/rapids/HostColumnarToGpu.scala

```diff
@@ -113,8 +115,24 @@ object HostColumnarToGpu {
         for (i <- 0 until rows) {
           b.appendUTF8String(cv.getUTF8String(i).getBytes)
         }
-      case (t, _) =>
-        throw new UnsupportedOperationException(s"Converting to GPU for $t is not currently " +
+      case (dt: DecimalType, nullable) =>
+        // Because DECIMAL64 is the only supported decimal DType, we can
+        // append unscaledLongValue instead of BigDecimal itself to speedup this conversion.
+        if (nullable) {
+          for (i <- 0 until rows) {
+            if (cv.isNullAt(i)) {
+              b.appendNull()
+            } else {
+              b.append(cv.getDecimal(i, DType.DECIMAL64_MAX_PRECISION, dt.scale).toUnscaledLong)
+            }
+          }
+        } else {
+          for (i <- 0 until rows) {
+            b.append(cv.getDecimal(i, DType.DECIMAL64_MAX_PRECISION, dt.scale).toUnscaledLong)
+          }
+        }
+      case (t, n) =>
+        throw new UnsupportedOperationException(s"Converting to GPU for ${t} is not currently " +
           s"supported")
```

Comment:
also, not related to this diff, but while you are at it:
```suggestion
          "supported")
```